hardware_interface/include/hardware_interface/types/hardware_interface_error_signals.hpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HARDWARE_INTERFACE__TYPES__HARDWARE_INTERFACE_ERROR_SIGNALS_HPP_
+#define HARDWARE_INTERFACE__TYPES__HARDWARE_INTERFACE_ERROR_SIGNALS_HPP_
+
+#include <cstdint>
+#include <vector>
+
+namespace hardware_interface
+{
+constexpr char ERROR_SIGNAL_INTERFACE_NAME[] = "ERROR_SIGNAL";
+// Available error signal names
+enum class error_signal : std::uint8_t
```

Comment:
The idea is for later if we support variant to set the vector at position hardware_interface::error_signal::ERROR_SIGNAL_0 to the error code